### PR TITLE
TE-1033: Allows the vertical spacing to be optional in `Amenities`

### PR DESCRIPTION
**Pull request by stephenkr**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1033)

### What **one** thing does this PR do?
Allows the vertical spacing to be optional in `Amenities`

### Other Notes
- Since Amenities may exist nested in another component, we need to remove the spacing
- Since Amenities can appear as a primary component in our pages, we need to keep the padding

__Before__
screen shot 2018-09-10 at 12 22 50

__After__
screen shot 2018-09-10 at 12 22 41

### CODE REVIEW

File: src/components/property-page-widgets/Amenities/Readme.md [hunk 1]
```diff
@@ -242,3 +242,29 @@ const amenities = [
...
+  },
+  {
+    name: 'Outside',
+    iconName: 'sun',
+    items: ['Tennis Court'],
+  }
+];
+
+
+<Amenities isNested amenities={amenities} headingText="Property Amenities" />
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Amenities/Readme.md]: The prop name `isNested` couples this use case to the component API. 

There are a bunch of reasons why you might not want the vertical gutters, so let's name the prop according to what it does.

I suggested `hasVerticalGutters`, defaulting to `true`

File: src/components/property-page-widgets/Amenities/Readme.md [hunk 2]
```diff
@@ -242,3 +242,29 @@ const amenities = [
 <Amenities hasExtraItemsInModal amenities={amenities} headingText="Property Amenities" />
 
 ```
+
+#### Remove vertical padding
```
[the remark this answers is not in the document]
Comment (stephenkr) [reply src/components/property-page-widgets/Amenities/Readme.md]: Good idea, i'll change this.

File: src/components/property-page-widgets/Amenities/Readme.md (hunk 1 above)
Comment (stephenkr) [reply src/components/property-page-widgets/Amenities/Readme.md]: Ah good idea, i'll change the prop name

File: src/components/property-page-widgets/Amenities/utils/getAmenityMarkup.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { getCategoryMarkup } from './getCategoryMarkup';
+import { getExtraItemsMarkup } from './getExtraItemsMarkup';
+
+/**
+ * @param {Object[]}  amenities
+ * @param {boolean}   hasExtraItemsInModal
+ * @param {string}    headingText
+ * @param {boolean}   isStacked
+ * @param {string}    modalTriggerText
+ */
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Amenities/utils/getAmenityMarkup.js]: Please add a `@returns` tag.

Comment (stephenkr) [reply src/components/property-page-widgets/Amenities/utils/getAmenityMarkup.js]: Fixing now.

**Pull request merged**